### ISSUE TITLE
debug_traceTransaction Error: "tracing failed: insufficient funds for transfer: address"

### ISSUE THREAD

**Issue by keennay** [kcc-community/kcc#24]
KCC off commit: https://github.com/kcc-community/kcc/commit/1c1715bd00c6e9caff7c7fc33c55025e362e439c sees the following error when attempting to call debug_traceTransaction on certain hashes: `tracing failed: insufficient funds for transfer: address`

Example:

`curl -X POST --data '{"jsonrpc":"2.0","method":"debug_traceTransaction","params":["0x934e3a7564d57dcf8880d8f7c2fc013512edc6027e157cc3352037f650c5c9c8", {"tracer": "callTracer"}],"id":1}' -H "Content-Type: application/json" http://127.0.0.1:8545`

`{"jsonrpc":"2.0","id":1,"error":{"code":-32000,"message":"tracing failed: insufficient funds for transfer: address 0x02C8B0d968e8f60cF5A079eC0c9b443fa6C53e24"}}`

**Comment by junnmm** [comment]
Hi, ymonye 

Thanks for your reporting.   
We will check it.

**Comment by junnmm** [comment]
@0xcary  

Seems to be a bug in the tracer:

I think this should not happen. Before the tx of "0x934e3a7564d57dcf8880d8f7c2fc013512edc6027e157cc3352037f650c5c9c8", the user has more than 4 KCS in his balance.   


```
Balance of 0x02C8B0d968e8f60cF5A079eC0c9b443fa6C53e24 at #1069378: 14.542071867491999638
Balance of 0x02C8B0d968e8f60cF5A079eC0c9b443fa6C53e24 at #1069379: 4.537672192491999638

bal after tx 0x3ad275b290c32c6370ad30df0eb1deacacb1a0008368ae27847306198d209b5f: 12.541792207491999638
bal after tx 0x44a80e93fe8c624f5f5b4bdd102046b78610d545695597a3866f69673c9b4283: 10.538995607491999638
bal after tx 0xba908560f9269ccdeefe1afc8608fb45a96f412902c9d6cd016d186c04b3d480: 8.538967641491999638
bal after tx 0xe98895d97686445012074255d9e060bf490ce7222001b4a6ec44429790a3db97: 8.538795344491999638
bal after tx 0x4c7c78866503e7e7895af54ef9343419247221056475cfa32924e8f5da6d9eb8: 8.538650563491999638
bal after tx 0x1b701f56601b815140be18ebd4e7334265931978b65f54efd847a3dde49d7f5f: 8.538505782491999638
bal after tx 0xcdda16dbae85eafcfc190098d6a8a91027c5dc4ff61f519b944789475aa01238: 8.538333485491999638
bal after tx 0x90e152b21f37391ab4fedbda08dc07b2bf29b0b52687253d4684df329a7b4d35: 8.538188704491999638
bal after tx 0x02a719f3cec253e92d1b4d58b10b17767420ad74a64467ab2d2e4c525226f49d: 8.538043923491999638
bal after tx 0x32deaa4fea8f76f285143da89e5ff6c723aabc168d0251fba2a99edb18595d5f: 8.537837434491999638
bal after tx 0x4e8d4789fbcb190a5b7e224c55381b97ec1f1be0d1ca6e804fd1494008b9866b: 6.537809468491999638
bal after tx 0xa68cf22ca6bb3c208c8ba7a03cbef32ca81080655ba6640813702f3053f6e468: 4.537781502491999638
bal after tx 0x5c0184d065d565dfb8bdf1ff6c36e0a37d125d5c475afc4cd0e6bcead7d4a082: 4.537755536491999638
bal after tx 0x3ff317040864aeafd5b9661371bcd596a83c15b78fe196bc2adfbcfbb37c6063: 4.537724124491999638
bal after tx 0xbffa3b189f23bf1c3a293c6bb18b898d77dc53ae89473fb848a74d91ad7c9f17: 4.537698158491999638
bal after tx 0x934e3a7564d57dcf8880d8f7c2fc013512edc6027e157cc3352037f650c5c9c8: 4.537672192491999638
```

May be related to :

- https://github.com/ethereum/go-ethereum/pull/22473  
- https://github.com/ethereum/go-ethereum/issues/22475

**Comment by keennay** [comment]
> @0xcary
> 
> Seems to be a bug in the tracer:
> 
> I think this should not happen. Before the tx of "0x934e3a7564d57dcf8880d8f7c2fc013512edc6027e157cc3352037f650c5c9c8", the user has more than 4 KCS in his balance.
> 
> ```
> Balance of 0x02C8B0d968e8f60cF5A079eC0c9b443fa6C53e24 at #1069378: 14.542071867491999638
> Balance of 0x02C8B0d968e8f60cF5A079eC0c9b443fa6C53e24 at #1069379: 4.537672192491999638
> 
> bal after tx 0x3ad275b290c32c6370ad30df0eb1deacacb1a0008368ae27847306198d209b5f: 12.541792207491999638
> bal after tx 0x44a80e93fe8c624f5f5b4bdd102046b78610d545695597a3866f69673c9b4283: 10.538995607491999638
> bal after tx 0xba908560f9269ccdeefe1afc8608fb45a96f412902c9d6cd016d186c04b3d480: 8.538967641491999638
> bal after tx 0xe98895d97686445012074255d9e060bf490ce7222001b4a6ec44429790a3db97: 8.538795344491999638
> bal after tx 0x4c7c78866503e7e7895af54ef9343419247221056475cfa32924e8f5da6d9eb8: 8.538650563491999638
> bal after tx 0x1b701f56601b815140be18ebd4e7334265931978b65f54efd847a3dde49d7f5f: 8.538505782491999638
> bal after tx 0xcdda16dbae85eafcfc190098d6a8a91027c5dc4ff61f519b944789475aa01238: 8.538333485491999638
> bal after tx 0x90e152b21f37391ab4fedbda08dc07b2bf29b0b52687253d4684df329a7b4d35: 8.538188704491999638
> bal after tx 0x02a719f3cec253e92d1b4d58b10b17767420ad74a64467ab2d2e4c525226f49d: 8.538043923491999638
> bal after tx 0x32deaa4fea8f76f285143da89e5ff6c723aabc168d0251fba2a99edb18595d5f: 8.537837434491999638
> bal after tx 0x4e8d4789fbcb190a5b7e224c55381b97ec1f1be0d1ca6e804fd1494008b9866b: 6.537809468491999638
> bal after tx 0xa68cf22ca6bb3c208c8ba7a03cbef32ca81080655ba6640813702f3053f6e468: 4.537781502491999638
> bal after tx 0x5c0184d065d565dfb8bdf1ff6c36e0a37d125d5c475afc4cd0e6bcead7d4a082: 4.537755536491999638
> bal after tx 0x3ff317040864aeafd5b9661371bcd596a83c15b78fe196bc2adfbcfbb37c6063: 4.537724124491999638
> bal after tx 0xbffa3b189f23bf1c3a293c6bb18b898d77dc53ae89473fb848a74d91ad7c9f17: 4.537698158491999638
> bal after tx 0x934e3a7564d57dcf8880d8f7c2fc013512edc6027e157cc3352037f650c5c9c8: 4.537672192491999638
> ```
> 
> May be related to :
> 
> * [eth, les: fix tracer ethereum/go-ethereum#22473](https://github.com/ethereum/go-ethereum/pull/22473)
> * [debug_traceTransaction gas usage doesn't match transaction receipt ethereum/go-ethereum#22475](https://github.com/ethereum/go-ethereum/issues/22475)

Thanks, it appears that was the particular fix needed to get past that error. Made the edits within commit https://github.com/kcc-community/kcc/commit/1c1715bd00c6e9caff7c7fc33c55025e362e439c and rebuilt locally:

`curl -X POST --data '{"jsonrpc":"2.0","method":"debug_traceTransaction","params":["0x934e3a7564d57dcf8880d8f7c2fc013512edc6027e157cc3352037f650c5c9c8", {"tracer": "callTracer"}],"id":1}' -H "Content-Type: application/json" http://127.0.0.1:8545`

`{"jsonrpc":"2.0","id":1,"result":{"type":"CALL","from":"0x02c8b0d968e8f60cf5a079ec0c9b443fa6c53e24","to":"0x4446fc4eb47f2f6586f9faab68b3498f86c07521","value":"0x1bc16d674ec80000","gas":"0x1326","gasUsed":"0x1326","input":"0xd0e30db0","error":"out of gas","time":"1.852519ms"}}`

**Comment by junnmm** [comment]
> > @0xcary
> > Seems to be a bug in the tracer:
> > I think this should not happen. Before the tx of "0x934e3a7564d57dcf8880d8f7c2fc013512edc6027e157cc3352037f650c5c9c8", the user has more than 4 KCS in his balance.
> > ```
> > Balance of 0x02C8B0d968e8f60cF5A079eC0c9b443fa6C53e24 at #1069378: 14.542071867491999638
> > Balance of 0x02C8B0d968e8f60cF5A079eC0c9b443fa6C53e24 at #1069379: 4.537672192491999638
> > 
> > bal after tx 0x3ad275b290c32c6370ad30df0eb1deacacb1a0008368ae27847306198d209b5f: 12.541792207491999638
> > bal after tx 0x44a80e93fe8c624f5f5b4bdd102046b78610d545695597a3866f69673c9b4283: 10.538995607491999638
> > bal after tx 0xba908560f9269ccdeefe1afc8608fb45a96f412902c9d6cd016d186c04b3d480: 8.538967641491999638
> > bal after tx 0xe98895d97686445012074255d9e060bf490ce7222001b4a6ec44429790a3db97: 8.538795344491999638
> > bal after tx 0x4c7c78866503e7e7895af54ef9343419247221056475cfa32924e8f5da6d9eb8: 8.538650563491999638
> > bal after tx 0x1b701f56601b815140be18ebd4e7334265931978b65f54efd847a3dde49d7f5f: 8.538505782491999638
> > bal after tx 0xcdda16dbae85eafcfc190098d6a8a91027c5dc4ff61f519b944789475aa01238: 8.538333485491999638
> > bal after tx 0x90e152b21f37391ab4fedbda08dc07b2bf29b0b52687253d4684df329a7b4d35: 8.538188704491999638
> > bal after tx 0x02a719f3cec253e92d1b4d58b10b17767420ad74a64467ab2d2e4c525226f49d: 8.538043923491999638
> > bal after tx 0x32deaa4fea8f76f285143da89e5ff6c723aabc168d0251fba2a99edb18595d5f: 8.537837434491999638
> > bal after tx 0x4e8d4789fbcb190a5b7e224c55381b97ec1f1be0d1ca6e804fd1494008b9866b: 6.537809468491999638
> > bal after tx 0xa68cf22ca6bb3c208c8ba7a03cbef32ca81080655ba6640813702f3053f6e468: 4.537781502491999638
> > bal after tx 0x5c0184d065d565dfb8bdf1ff6c36e0a37d125d5c475afc4cd0e6bcead7d4a082: 4.537755536491999638
> > bal after tx 0x3ff317040864aeafd5b9661371bcd596a83c15b78fe196bc2adfbcfbb37c6063: 4.537724124491999638
> > bal after tx 0xbffa3b189f23bf1c3a293c6bb18b898d77dc53ae89473fb848a74d91ad7c9f17: 4.537698158491999638
> > bal after tx 0x934e3a7564d57dcf8880d8f7c2fc013512edc6027e157cc3352037f650c5c9c8: 4.537672192491999638
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > May be related to :
> > 
> > * [eth, les: fix tracer ethereum/go-ethereum#22473](https://github.com/ethereum/go-ethereum/pull/22473)
> > * [debug_traceTransaction gas usage doesn't match transaction receipt ethereum/go-ethereum#22475](https://github.com/ethereum/go-ethereum/issues/22475)
> 
> Thanks, it appears that was the particular fix needed to get past that error. Made the edits within commit [1c1715b](https://github.com/kcc-community/kcc/commit/1c1715bd00c6e9caff7c7fc33c55025e362e439c) and rebuilt locally:
> 
> `curl -X POST --data '{"jsonrpc":"2.0","method":"debug_traceTransaction","params":["0x934e3a7564d57dcf8880d8f7c2fc013512edc6027e157cc3352037f650c5c9c8", {"tracer": "callTracer"}],"id":1}' -H "Content-Type: application/json" http://127.0.0.1:8545`
> 
> `{"jsonrpc":"2.0","id":1,"result":{"type":"CALL","from":"0x02c8b0d968e8f60cf5a079ec0c9b443fa6c53e24","to":"0x4446fc4eb47f2f6586f9faab68b3498f86c07521","value":"0x1bc16d674ec80000","gas":"0x1326","gasUsed":"0x1326","input":"0xd0e30db0","error":"out of gas","time":"1.852519ms"}}`

Great!  

We will leave this issue open and try to fix it in the next release.
